include/aws/http/proxy.h

```diff
@@ -506,6 +506,20 @@ struct aws_http_proxy_config *aws_http_proxy_config_new_from_proxy_options(
     struct aws_allocator *allocator,
     const struct aws_http_proxy_options *options);
 
+/**
+ * Create a persistent proxy configuration from non-persistent proxy options.
+ *
+ * @param allocator memory allocator to use
+ * @param options http proxy options to source proxy configuration from
+ * @param tls_connection_options tls connection options to determine connection_type if connection type is legacy
```

Comment:
Why not expose `s_determine_proxy_connection_type` instead?

Reply:
`s_determine_proxy_connection_type` is an implementation detail. This gives us the the flexibility to change implementation later without caring about breaking consumers.

Reply:
The whole point of using `tls_connection_options` is to determine what connection type is. Just have a simple if/else maybe better.

Reply:
I have updated the `s_determine_proxy_connection_type` to use a bool instead so that we can have simple if/else. If we make expose `s_determine_proxy_connection_type`, we will have to expose `s_aws_http_proxy_config_new` as well.